src/collection.js

```diff
@@ -104,8 +106,14 @@ export default class Collection {
     });
   }
 
-  use(transfomer) {
-    this._transformers[transfomer.type].push(transfomer);
+  use(plugin) {
```

Comment:
It seems that this brings a situation where different extensibility mechanisms (e.g. plugins, adapters, transformers) overlaps, without a clear distinction in their names.

Until now, we had a clear scoping with each of the concepts, if we introduce "plugins", I fear that we might lose this. Should we consider `generators` (like we did for Kinto server) ?